make.ps1

```diff
@@ -0,0 +1,235 @@
+{
+	if (Test-Path ./engine/OpenRA.Utility.exe)
+	{
+		$msg = "Testing " + $modID + " mod MiniYAML"
+		echo $msg
+		./engine/OpenRA.Utility.exe $modID --check-yaml
+	}
+	else
+	{
+		UtilityNotFound
+	}
+}
+
+function Check-Command
+{
+	if (Test-Path ./engine/OpenRA.Utility.exe)
+	{
+		echo "Checking for explicit interface violations..."
+		./engine/OpenRA.Utility.exe all --check-explicit-interfaces
+
+		$msg = "Checking for code style violations in OpenRA.Mods." + $modID + "..."
+		echo $msg
+		./engine/OpenRA.Utility.exe $modID --check-code-style OpenRA.Mods.$modID
+	}
+	else
+	{
+		UtilityNotFound
+	}
+}
+
+function Check-Scripts-Command
+{
+	if ((Get-Command "luac.exe" -ErrorAction SilentlyContinue) -ne $null)
+	{
+		echo "Testing Lua scripts..."
+		foreach ($script in ls "mods/*/maps/*/*.lua")
+		{
+			luac -p $script
+		}
+		echo "Check completed!"
+	}
+	else
+	{
+		echo "luac.exe could not be found. Please install Lua."
+	}
+}
+
+function Docs-Command
+{
+	if (Test-Path ./engine/OpenRA.Utility.exe)
+	{
+		Version-Command
+		./engine/OpenRA.Utility.exe all --docs | Out-File -Encoding "UTF8" DOCUMENTATION.md
+		./engine/OpenRA.Utility.exe all --lua-docs | Out-File -Encoding "UTF8" Lua-API.md
+	}
+	else
+	{
+		UtilityNotFound
+	}
+}
+
+function FindMSBuild
+{
+	$msBuildVersions = @("4.0")
+	foreach ($msBuildVersion in $msBuildVersions)
+	{
+		$key = "HKLM:\SOFTWARE\Microsoft\MSBuild\ToolsVersions\{0}" -f $msBuildVersion
+		$property = Get-ItemProperty $key -ErrorAction SilentlyContinue
+		if ($property -eq $null -or $property.MSBuildToolsPath -eq $null)
+		{
+			continue
+		}
+		$path = Join-Path $property.MSBuildToolsPath -ChildPath "MSBuild.exe"
+		if (Test-Path $path)
+		{
+			return $path
+		}
+	}
+	return $null
+}
+
+function UtilityNotFound
+{
+	echo "OpenRA.Utility.exe could not be found. Build the project first using the `"all`" command."
+}
+###############################################################
+############################ Main #############################
+###############################################################
+if ($args.Length -eq 0)
+{
+	echo "Command list:"
+	echo ""
+	echo "  all             Builds the game, its development tools and the mod dlls."
+	echo "  version	        Sets the version strings for all mods to the latest"
+	echo "                  version for the current Git branch."
+	echo "  clean           Removes all built and copied files."
+	echo "                  from the mods and the engine directories."
+	echo "  test            Tests the mod's MiniYAML for errors."
+	echo "  check           Checks .cs files for StyleCop violations."
+	echo "  check-scripts   Checks .lua files for syntax errors."
+	echo "  docs            Generates the trait and Lua API documentation."
+	echo ""
+	$command = (Read-Host "Enter command").Split(' ', 2)
+}
+else
+{
+	$command = $args
+}
+
+if ($command -ne "docs" -and $command -ne "test")
+{
+	#Run the same command on the engine's make file
+	cd ".\engine\"
```

Comment:
This starts a cascade of failures if the engine submodule hasn't been initialized.

Please add a check that gives an error prompting the user to install the engine files, either using `git submodule init` and `git submodule update` or manually if they aren't using git.